### PR TITLE
fix: Get rid of remaining hydration errors

### PR DESCRIPTION
**Pull request by chargome**
fixes [#11070](https://github.com/getsentry/sentry-docs/issues/11070)

### CODE REVIEW

**Review by a-hariti — APPROVED** [review]

File: src/components/codeContext.tsx [hunk 1]
```diff
@@ -277,10 +284,21 @@ export async function createOrgAuthToken({
...
+  }
+  return {};
+};
+
 export function CodeContextProvider({children}: {children: React.ReactNode}) {
   const [codeKeywords, setCodeKeywords] = useState(cachedCodeKeywords ?? DEFAULTS);
-
   const [isLoading, setIsLoading] = useState<boolean>(cachedCodeKeywords ? false : true);
+  const [storedCodeSelection, setStoredCodeSelection] = useState<SelectedCodeTabs>(() =>
+    getLocallyStoredSelections()
```
Comment (lforst) [inline src/components/codeContext.tsx:300]: l/m: As discussed: I would not make the initial state dependent on localStorage as it invites hydration errors in the future.

**Review by lforst — APPROVED** [review]
nice!

File: src/components/codeContext.tsx (hunk 1 above)
Comment (chargome) [reply src/components/codeContext.tsx]: updated it 👍

File: src/components/codeContext.tsx [hunk 2]
```diff
@@ -277,10 +284,24 @@ export async function createOrgAuthToken({
...
 export function CodeContextProvider({children}: {children: React.ReactNode}) {
   const [codeKeywords, setCodeKeywords] = useState(cachedCodeKeywords ?? DEFAULTS);
-
   const [isLoading, setIsLoading] = useState<boolean>(cachedCodeKeywords ? false : true);
+  const [storedCodeSelection, setStoredCodeSelection] = useState<SelectedCodeTabs>({});
+
+  // populate state using localstorage
+  useEffect(() => {
+    setStoredCodeSelection(getLocallyStoredSelections());
+  }, []);
```
Comment (lforst) [inline src/components/codeContext.tsx:304]: with this I think we can get rid of the isLocalStorageAvailable check right?

Comment (chargome) [reply src/components/codeContext.tsx:304]: I would just leave it tbh

Comment (lforst) [reply src/components/codeContext.tsx:304]: sounds good

**Pull request merged**